### PR TITLE
fix NodeCidr logic issue

### PR DESCRIPTION
**Pull request by neo-liang-sap**

### CODE REVIEW

**Comment by rakesh-garimella** [comment]
Hi @neo-liang-sap  I looked into the code and indeed cidr mask is not being sent. Try doing following
```-       if v, ok := in["node_cidr_mask_size"].(*int32); ok {
-               obj.NodeCIDRMaskSize = v
+       if v, ok := in["node_cidr_mask_size"].(int); ok {
+               s := int32(v)
+.              obj.NodeCIDRMaskSize = &s
+       }

**Comment by neo-liang-sap** [comment]
@rakesh-garimella yes it's working now, i've updated the PR

File: terraform-provider-gardener.code-workspace [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+	"folders": [
+		{
+			"path": "."
+		}
+	],
+	"settings": {}
+}
```
Comment (rakesh-garimella) [inline terraform-provider-gardener.code-workspace]: i guess  we dont need this ?

**Comment by rakesh-garimella** [comment]
Hi @neo-liang-sap  there is a binary file also commited can you please remove that also can you add the tests too ? Also when ready please remove WIP label :)

**Comment by neo-liang-sap** [comment]
Hi @rakesh-garimella , thanks for catching out and sorry i mistake add two files locally to PR.
I've fixed the errors.

test case added

**Comment by rakesh-garimella** [comment]
Tried with following
```
    kubernetes {
      version = "1.17.3"
      kube_controller_manager {
        node_cidr_mask_size = "22"
      }
    }
```
But still I get the default cidr 24 set
a working terrform template could be useful. @neo-liang-sap  can you provide one here so that I can test and merge it if things work.

**Comment by rakesh-garimella** [comment]
I would close this PR because of inactivity feel free to open a new PR also please add a working example which I can use to test the PR.

**Pull request closed without merging**